zanata-common-api/src/main/java/org/zanata/rest/dto/GlossaryEntry.java

```diff
@@ -51,11 +52,63 @@
     */
     private static final long serialVersionUID = 1685907304736580890L;
 
+    private String resId;
+
+    private String pos;
+
+    private String description;
+
     private List<GlossaryTerm> glossaryTerms;
 
     private LocaleId srcLang;
 
-    private String sourcereference;
+    private String sourceReference;
+
+    private int termsCount = 0;
+
+    public GlossaryEntry() {
+        this(null);
+    }
+
+    public GlossaryEntry(String resId) {
+        this.resId = resId;
+    }
+
+    @XmlElement(name = "resId", namespace = Namespaces.ZANATA_OLD)
+    public String getResId() {
+        return resId;
+    }
+
+    public void setResId(String resId) {
+        this.resId = resId;
+    }
+
+    @XmlElement(name = "pos", namespace = Namespaces.ZANATA_OLD)
+    public String getPos() {
+        return pos;
+    }
+
+    public void setPos(String pos) {
+        this.pos = pos;
+    }
+
+    @XmlElement(name = "description", namespace = Namespaces.ZANATA_OLD)
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    @XmlElement(name = "termsCount", namespace = Namespaces.ZANATA_OLD)
```

Comment:
Use `Namespaces.ZANATA_API` for this property, and the other new properties above.